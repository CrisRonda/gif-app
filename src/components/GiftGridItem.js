import React from "react";
import PropTypes from "prop-types";
import { Card, ButtonIcon } from "react-rainbow-components";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faShareAlt,
  faAirFreshener as faHeart,
} from "@fortawesome/free-solid-svg-icons";

const Footer = () => (
  <div className="rainbow-align-content_space-between">
    <div className="rainbow-flex">
      <ButtonIcon
        icon={<FontAwesomeIcon icon={faHeart} />}
        className="rainbow-m-right_xx-small"
      />
      <ButtonIcon icon={<FontAwesomeIcon icon={faShareAlt} />} />
    </div>
  </div>
);

const GiftGridItem = ({ title, url, small, medium, hideImage }) => {
  let style = { margin: "24px", maxWidth: 600 };
  if (small) {
    style = { margin: "24px", maxWidth: 100 };
  }
  if (medium) {
    style = { margin: "24px", maxWidth: 450 };
  }
  return (
    <Card
      title={title}
      style={{
        // ...style,
        display: "flex",
        flexDirection: "column",
        justifyContent: "space-between",
      }}
      className={"card animate__animated animate__fadeIn"}
    >
      <figure>
        {!hideImage && (
          <img style={{ flex: 1, width: "100%" }} src={url} alt={title} />
        )}
        <figcaption>{title}</figcaption>
      </figure>
    </Card>
  );
};
GiftGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  small: PropTypes.bool,
  medium: PropTypes.bool,
  hideImage: PropTypes.bool,
};
GiftGridItem.defaultProps = {
  small: false,
  medium: false,
  hideImage: false,
};
export { GiftGridItem, Footer };
export default GiftGridItem;
