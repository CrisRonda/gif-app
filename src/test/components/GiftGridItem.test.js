import React from "react";
import { shallow } from "enzyme";
import GiftGridItem from "../../components/GiftGridItem";

describe("test on GiftGridItem", () => {
  const title = "My title.. ";
  const href = "https://localhost/";

  const wrapper = shallow(<GiftGridItem title={title} url={href} />);
  // test("should render GiftGridItem", () => {
  //   expect(wrapper).toMatchSnapshot();
  // });

  test("should have alt and src on img", () => {
    const img = wrapper.find("img");
    const { src, alt } = img.props();
    expect(src).toBe(href);
    expect(alt).toBe(title);
  });

  // test("should have class animate__fadeIn", () => {
  //   const container = wrapper.find("figure");
  //   const { className } = container.props();
  //   expect(className.includes("animate__fadeIn")).toBe(false);
  // });
});
