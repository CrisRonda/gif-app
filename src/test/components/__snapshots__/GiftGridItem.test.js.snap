// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on GiftGridItem should render GiftGridItem 1`] = `
<figure
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="My title.. "
    src="https://localhost/"
  />
  <figcaption>
    My title.. 
  </figcaption>
</figure>
`;
